k8s_handle/__init__.py

```diff
@@ -161,6 +169,8 @@ def _handler_provision(command, resources, priority_evaluator, use_kubeconfig, s
                                  help='Check existence of all env variables in config.yaml and stop if var is not set')
 parser_provisioning.add_argument('--use-kubeconfig', action='store_true', required=False,
                                  help='Try to use kube config')
+parser_provisioning.add_argument('--in-cluster', action='store_true', required=False,
```

Comment:
It think that if kube-config configuration and in-cluster configuration options are mutually exclusive by idea and by design, they should be mutually exclusive at arguments parser's level as well.

Now we can specify both and the kub-econfig will be implicitly used.
As a little effort, I'd try to add them into add_mutually_exclusive_group(), if it works correctly with subparsers.